liquibase-core/src/main/java/liquibase/command/CommandScope.java

```diff
@@ -40,7 +41,12 @@ public class CommandScope {
      * Creates a new scope for the given command.
      */
     public CommandScope(String... commandName) throws CommandExecutionException {
-        setOutput(System.out);
+        /*
+        This is an UncloseableOutputStream because we do not want individual command steps to inadvertently (or
+        intentionally) close the System.out OutputStream. Closing System.out renders it unusable for other command
+        steps which expect it to still be open.
+         */
+        setOutput(new UnclosableOutputStream(System.out));
```

Comment:
I think the `new UnclosableOutputStream` call should be in the `setOutput()` function instead of here. No matter what people pass in as an output, it should be unclosable.

Reply:
I can do that. That felt a bit more assertive to me, but I can also see why that would be useful, since we wouldn't want our code closing a stream that was given to us.